### PR TITLE
Signals

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
cc @gkoz

### CODE REVIEW

File: src/signal.rs [hunk 1]
```diff
@@ -2305,3 +2306,154 @@ mod cell_renderer {
...
+
+mod container {
+    use std::mem::transmute;
+    use glib::signal::connect;
+    use glib::translate::*;
+    use ffi::{GtkContainer, GtkWidget};
+    use {Container, Widget};
+
+    impl super::ContainerSignals for Container {
+        fn connect_add<F: Fn(&Self, &Widget) + 'static>(&self, f: F) -> u64 {
```
Comment (GuillaumeGomez) [inline src/signal.rs]: I'm not sure here if `Widget` shouldn't be `IsA<Widget>`.

**Comment by EPashkin** [comment]
https://developer.gnome.org/gtk3/stable/GtkCellRendererText.html ,
https://developer.gnome.org/gtk3/stable/GtkCheckMenuItem.html ,
https://developer.gnome.org/gtk3/stable/GtkContainer.html ,
have children, so `<T>`

**Comment by EPashkin** [comment]
`GtkContainer` signals can be generated https://gist.github.com/EPashkin/977773a192bf633bb8413a96a1930870, but currently don't contains `IsWidget<T>` (IMHO needed)

**Comment by EPashkin** [comment]
..(forgot add main idea), so if `IsA<Widget>` not needed and no need in this signals currently, IMHO not need add signals as it simply.

**Comment by gkoz** [comment]
`IsA<Widget>` couldn't work: the handler can't assume anything about the types of the child widgets, we can't predict anything statically.
More generally, I suspect that the container signals are not really supposed to be used by applications. Note the absence of documentation. I haven't tested this but I have the impression that `Box` doesn't emit any of `Container` signals (e.g. packing a widget might not emit the `add` signal).

**Comment by GuillaumeGomez** [comment]
@gkoz: Is it really an issue? At least we'll have them *if* anyone needs them.

@EPashkin: I updated the PR, do you see anything else? (I made `CellRendererText`, `CheckMenuItem` and `Container` "serializable".)

**Comment by EPashkin** [comment]
Thanks, sees it all ok.

**Comment by gkoz** [comment]
> Is it really an issue? At least we'll have them if anyone needs them.

If someone needs them but they don't match reasonable expectations it is an issue. Can you test with a few containers to see if `add` and `remove` are fired as expected?

Regarding someone needing any obscure signal, I think it might be possible to make a dynamically-typed interface for connecting any signals.

**Comment by GuillaumeGomez** [comment]
@gkoz: It's working (tested on VerticalLayout in process-viewer [here](https://github.com/GuillaumeGomez/process-viewer/tree/test_signal)). :)

**Comment by GuillaumeGomez** [comment]
So, can we merge it?

**Comment by gkoz** [comment]
Thanks for testing these signals @GuillaumeGomez

**Pull request merged**